### PR TITLE
feat: refactor and dockerize credere

### LINKED ISSUE #469: Consider a credere_common Pillar file
To have less repetition across pillar/credere.sls and credere_dev.sls (similar to what cove.sls does)

This can include common `apache` configuration and `docker_apps.env` variables, etc.

### PR DESCRIPTION
**Pull request by yolile**
closes #469

### CODE REVIEW

File: pillar/credere_common.sls [hunk 1]
```diff
@@ -1,57 +1,43 @@
-network:
-  host_id: ocp24
-  ipv4: 213.52.130.126
-  ipv6: "2a01:7e00:e000:06e6::"
-  networkd:
-    template: linode
-    gateway4: 213.52.130.1
```
Comment (jpmckinney) [inline pillar/credere_common.sls:7]: Missing from credere_production

File: pillar/credere_common.sls [hunk 2]
```diff
@@ -1,57 +1,43 @@
...
-      servername: ocp24.open-contracting.org
       # servername: credere.open-contracting.org
       context:
         proxypass: http://localhost:8000/
         # authname: Credere Staging
 
 docker:
   user: deployer
+  docker_compose:
+    version: v2.19.0
```
Comment (jpmckinney) [inline pillar/credere_common.sls:14]: This should only be on credere_dev (if even there). Docker Compose is now part of Docker itself, and the production server doesn't install it separately.

File: pillar/credere_common.sls [hunk 3]
```diff
@@ -1,57 +1,43 @@
...
 apache:
   public_access: True
   sites:
     credere:
       configuration: proxy
-      servername: ocp24.open-contracting.org
       # servername: credere.open-contracting.org
       context:
         proxypass: http://localhost:8000/
         # authname: Credere Staging
```
Comment (jpmckinney) [inline pillar/credere_common.sls:9]: Delete the two lines of commented-out configuration

File: pillar/credere_common.sls [hunk 4]
```diff
@@ -1,57 +1,43 @@
...
       VITE_DEFAULT_LANG: es
-      VITE_HOST: ocp24.open-contracting.org # credere.open-contracting.org
       VITE_LOCALE: es-CO
       VITE_MAX_FILE_SIZE_MB: 5
       VITE_MORE_INFO_OCP_URL: https://www.open-contracting.org/es/
+  credere_backend:
+    target: credere-backend
+    port: 3000
+    env:
+      EMAIL_TEMPLATE_LANG: en
```
Comment (jpmckinney) [inline pillar/credere_common.sls:32]: Change to `es`?

File: pillar/credere_common.sls [hunk 5]
```diff
@@ -1,57 +1,43 @@
...
       VITE_APP_VERSION: 0.1.5
       VITE_CURRENCY: COP
       VITE_DEFAULT_LANG: es
-      VITE_HOST: ocp24.open-contracting.org # credere.open-contracting.org
       VITE_LOCALE: es-CO
       VITE_MAX_FILE_SIZE_MB: 5
       VITE_MORE_INFO_OCP_URL: https://www.open-contracting.org/es/
+  credere_backend:
+    target: credere-backend
+    port: 3000
```
Comment (jpmckinney) [inline pillar/credere_common.sls:30]: Why 3000? We usually have 800# in other Docker Compose files.

File: pillar/credere_production.sls [hunk 6]
```diff
@@ -0,0 +1,28 @@
+apache:
+  sites:
+    credere:
+      servername: ocp24.open-contracting.org
+
+docker_apps:
+  credere_frontend:
+    target: credere-frontend
+    port: 8000
```
Comment (jpmckinney) [inline pillar/credere_production.sls:9]: Don't need to repeat port from _common (same in credere_dev and for credere_backend)

File: pillar/credere_common.sls (hunk 5 above)
Comment (yolile) [reply pillar/credere_common.sls:30]: I'm not sure, I just used the same config as the current docker-compose file in the server, but I will change it to 8000 as usual

File: pillar/credere_production.sls [hunk 7]
```diff
@@ -0,0 +1,28 @@
+apache:
+  sites:
+    credere:
+      servername: ocp24.open-contracting.org
```
[the remark this answers is not in the document]
Comment (yolile) [reply pillar/credere_production.sls:4]: Yes, please

File: salt/docker_apps/files/credere.yaml [hunk 8]
```diff
@@ -1,8 +1,18 @@
-# Placeholder for custom application
-version: "3"
+x-shared: &shared
+  env_file: .env
+  user: "{{ pillar.docker.uid }}:{{ pillar.docker.uid }}"
+  extra_hosts:
+    - "host.docker.internal:host-gateway"
+  restart: unless-stopped
+
 services:
```
Comment (jpmckinney) [inline salt/docker_apps/files/credere.yaml:8]: The services don't need any volumes for uploaded files? I don't remember how those are handled.

pelican-frontend does something like: https://github.com/open-contracting/pelican-frontend/blob/main/backend/core/settings.py#L223-L233

File: pillar/credere_production.sls [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      ENVIRONMENT: production
+      # Timeline
+      APPLICATION_EXPIRATION_DAYS: 7
+      DAYS_TO_CHANGE_TO_LAPSED: 14
+      DAYS_TO_ERASE_BORROWERS_DATA: 7
+      PROGRESS_TO_REMIND_STARTED_APPLICATIONS: 0.7
+      REMINDER_DAYS_BEFORE_EXPIRATION: 3
+      # Email templates
+      LINK_LINK: https://ocp24.open-contracting.org # https://credere.open-contracting.org
+      IMAGES_BASE_URL: https://ocp24.open-contracting.org/images # https://credere.open-contracting.org/images
```
Comment (jpmckinney) [inline pillar/credere_production.sls:34]: I've created the CNAME, so you can replace the ocp24 URLs with the value in the comment (and update the Apache servername).

When deploying, for SSL/TLS, remember to do steps 3 and 4 at https://ocdsdeploy.readthedocs.io/en/latest/develop/update/apache.html#acquire-ssl-certificates

File: pillar/credere_production.sls [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+apache:
+  sites:
+    credere:
+      servername: ocp24.open-contracting.org
+
+docker_apps:
+  credere_frontend:
+    target: credere-frontend
+    env:
+      # Vite
```
Comment (jpmckinney) [inline pillar/credere_production.sls:18]:
```suggestion
```

File: pillar/credere_dev.sls [hunk 11]
```diff
@@ -16,63 +16,27 @@ ssh:
...
-  docker_compose:
-    version: v2.19.0
 
 docker_apps:
-  credere:
-    target: credere
+  credere_frontend:
+    target: credere-frontend
+    env:
+      # Vite
```
Comment (jpmckinney) [inline pillar/credere_dev.sls:27]:
```suggestion
```

File: pillar/credere_common.sls [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+docker:
+  user: deployer
+
+docker_apps:
+  credere_frontend:
+    target: credere-frontend
+    port: 8000
+    env:
+      # Vite
```
Comment (jpmckinney) [inline pillar/credere_common.sls:17]:
```suggestion
```

**Review by jpmckinney — COMMENTED** [review]
Feel free to merge after making updates.

**Pull request merged**